File: easy/2220_minimum_bit_flips_to_convert_number/solution.cpp
class Solution {
public:
  int minBitFlips(int start, int goal) {
    int flips = 0;

    for (int _ = 0; _ < 32; ++_) {
      if ((start & 1) != (goal & 1))
        ++flips;

      start >>= 1;
      goal >>= 1;
    }

    return flips;
  }
};

File: easy/2220_minimum_bit_flips_to_convert_number/solution.py
class Solution:
    def minBitFlips(self, start: int, goal: int) -> int:
        startBin = bin(start)
        goalBin = bin(goal)

        flips = 0

        for i in range(1, min(len(startBin), len(goalBin)) - 1):
            if startBin[-i] != goalBin[-i]:
                flips += 1

        if len(startBin) > len(goalBin):
            flips += startBin[: -len(goalBin) + 2].count("1")
        else:
            flips += goalBin[: -len(startBin) + 2].count("1")

        return flips
